src/main/java/htsjdk/tribble/annotation/Strand.java

```diff
@@ -24,29 +24,102 @@
 package htsjdk.tribble.annotation;
 
 /**
- * Enum for strand, which can be encoded as string
+ * Enum for strand, which can be encoded as a string
  */
 public enum Strand {
-    POSITIVE("+"), NEGATIVE("-"), NONE("!");  // not really sure what we should do for the NONE Enum
 
     /**
-     * How we represent the strand information as text
+     * Represent the positive or forward strand.
      */
-    private String encoding;
-    Strand(String str) {
-        encoding = str;
+    POSITIVE('+'),
+    /**
+     * Represent the negative or reverse strand.
+     */
+    NEGATIVE('-'),
+
+    /**
+     * Denotes that an strand designation is not applicable
+     * or is unknown.
+     */
+    NONE('!');  // not really sure what we should do for the NONE Enum
+
+    /**
+     * How we represent the strand information as a single {@code char}.
+     */
+    private final char charEncoding;
+
+    /**
+     * How we represent the strand information as a {@link String}.
+     */
+    private final String stringEncoding;
+
+    Strand(final char ch) {
+        charEncoding = ch;
+        stringEncoding = String.valueOf(charEncoding);
     }
 
     /**
      * provide a way to take an encoding string, and produce a Strand
      * @param encoding the encoding string
      * @return a Strand object, if an appropriate one cannot be located an IllegalArg exception
+     * @deprecated please use {@link #decode} instead.
      */
-    public static Strand toStrand(String encoding) {
-        for (Strand st : Strand.values())
-            if (st.encoding.equals(encoding))
-                return st;
-        throw new IllegalArgumentException("Unable to match encoding to Strand enum for encoding string " + encoding);
+    @Deprecated
+    public static Strand toStrand(final String encoding) {
+        return decode(encoding);
     }
 
+    /**
+     * Returns the {@link Strand} that a {@code char} value represents.
+     * @param ch the char encoding for a Strand.
+     * @throws IllegalArgumentException if input char does not
+     *   represent a {@link #Strand} instance.
+     * @return never {@code null}, a value so that {@code decode(c).encodeAsChar() == c}.
+     */
+    public static Strand decode(final char ch) {
+        switch (ch) {
+            case '+': return POSITIVE;
```

Comment:
This duplicates the relationship already established in the enum definition. Instead you could write
```
for (Strand strand : values()) {
    if (strand.charEncoding == ch) {
        return strand;
    }
}
throw new ...
```

Reply:
yeah... I guess it could be kinda of faster with the switch and I am relaying on the unit-test to verify consistency at build-time. But at best is premature optimization so happy to revert to that.

Reply:
If it turns out to be a performance issue, you can build a static `Map` using the enum values and also avoid the data duplication. Given that there are only three enum members it seems unlikely that this will be a bottleneck.

Reply:
In fact the map is likely to carry too much overhead.

Reply:
I was curious so I wrote a short test. The test showed that switch is only slightly faster than the Map. Using a for loop is an order of magnitude slower than both the switch and map. So if you want to avoid duplication, Map seems to be the best solution.

https://gist.github.com/pshapiro4broad/1acabaa3fe74ac03007f15b989fe78a8

Reply:
Wow! Never thought of it... perhaps the culprit is values()... if one is allowed to change its contents then it makes sense that each invocation would instantiate a new array.

Reply:
If you are still curious and since you have the test code ready for action then you could try to statically cache the result of values().

Reply:
The code is in the gist link in my comment, I'll update it when I have a chance, but you can update it as well.

I would expect that `Enum.values()` is a static reference since all enum instances are known at the time the .class file is loaded. It would be easy to try out.

Reply:
Arrays are always mutable, so I'm pretty sure it will allocate a fresh one on every call to values.

Reply:
I have done the test and that is the cause of the bad performance. Caching it into an static final bring time to the par with other solutions.

Reply:
Good to know, thanks for checking this out.